cmd/flux/bootstrap_git.go

```diff
@@ -205,6 +201,15 @@ func bootstrapGitCmdRun(cmd *cobra.Command, args []string) error {
 		RecurseSubmodules: bootstrapArgs.recurseSubmodules,
 	}
 
+	// Modify the synchronization URL on GitRepository using the --ssh-hostname flag
+	if (bootstrapArgs.sshHostname != "") && (repositoryURL.Host != bootstrapArgs.sshHostname) {
```

Comment:
I think `bootstrapArgs.sshHostname` shall be evaluated only if `tokenAuth` is not specified.

Reply:
If the expected behavior is to modify the host on GitRepository, should not be evaluated in both cases? I mean, it is possible that the naming of the flag is not right or I missunderstood what is expected for that flag.

This is done this way because you should keep the port or change it in both cases, for example, in out company we have all ports changed, even the SSH ones

Reply:
The `bootstrapArgs.sshHostname` is used to specify an ssh host if it's different that the https one. We don't need  it when using only https.

With  `repositoryURL.Host = my-git-server.com:7990`,  `bootstrapArgs.sshHostname = my-git-server.com:7999` and the option `bootstrapArgs.tokenAuth = true`, I expect the sync to happen on `my-git-server.com:7990` and discard the `bootstrapArgs.sshHostname`.

Reply:
Knowing this, I could let the original code and just removing the line `repositoryURL.Host = repositoryURL.Hostname()` would do the trick, no? because it will keep the value for `repositoryURL.Host` and this value can come with port or without it from the `--url` flag

Reply:
Yes I think we can test that.

Reply:
Fixed. It seems to keep the port when --token-auth enabled by not changing the `repositoryURL.Host`.
I tested bootstraping with ssh too and it seems to work.
Could you please confirm the behavior?